janus_core/cli/preprocess.py

```diff
@@ -0,0 +1,69 @@
+"""Set up MLIP preprocessing commandline interface."""
+
+from pathlib import Path
+from typing import Annotated
+
+from typer import Option, Typer
+
+app = Typer()
+
+
+@app.command()
+def preprocess(
+    mlip_config: Annotated[
+        Path, Option(help="Configuration file to pass to MLIP CLI.")
+    ],
+    log: Annotated[Path, Option(help="Path to save logs to.")] = Path(
+        "preprocess-log.yml"
+    ),
+    tracker: Annotated[
+        bool, Option(help="Whether to save carbon emissions of calculation")
+    ] = True,
+    summary: Annotated[
+        Path,
+        Option(
+            help=(
+                "Path to save summary of inputs, start/end time, and carbon emissions."
+            )
+        ),
+    ] = Path("preprocess-summary.yml"),
+):
+    """
+    Convert training data to hdf5 by passing a configuration file to the MLIP's CLI.
+
+    Parameters
+    ----------
+    mlip_config : Path
+        Configuration file to pass to MLIP CLI.
+    log : Optional[Path]
+        Path to write logs to. Default is Path("preprocess-log.yml").
+    tracker : bool
+        Whether to save carbon emissions of calculation in log file and summary.
+        Default is True.
+    summary : Optional[Path]
+        Path to save summary of inputs, start/end time, and carbon emissions. Default
+        is Path("preprocess-summary.yml").
+    """
+    from janus_core.cli.utils import carbon_summary, end_summary, start_summary
+    from janus_core.training.preprocess import preprocess as run_preprocess
+
+    inputs = {"mlip_config": str(mlip_config)}
+
+    # Save summary information before preprocessing begins
+    start_summary(command="preprocess", summary=summary, inputs=inputs)
```

Comment:
Because it has an entry/exit like structure, not sure whether we might want to restructure this as a context manager if we haven't tried already.

Not for here, though.